File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/opengl/model/GLCircle.java
package br.com.insanitech.spritekit.opengl.model;

/**
 * Created by anderson on 7/3/15.
 */
public class GLCircle extends GLGeometry {
    private int points = 10;

    public GLCircle() {
        vertices = new float[(points + 1) * 3];
        for (int i = 3; i < (points + 1) * 3; i += 3) {
            double rad = (i * 360 / points * 3) * (3.14 / 180);
            vertices[i] = (float) Math.cos(rad);
            vertices[i + 1] = (float) Math.sin(rad);
            vertices[i + 2] = 0;
        }
        componentsPerVertice = 3;
        generateVertex();
    }

    public int getPointsCount() {
        return points;
    }
}

File: SpriteKitLib/src/main/java/br/com/insanitech/spritekit/opengl/model/GLCircle.kt
package br.com.insanitech.spritekit.opengl.model

/**
 * Created by anderson on 7/3/15.
 */
internal class GLCircle : GLGeometry() {
    var pointsCount = 10

    init {
        this.generateCircleVertices()
        this.generateVertex()
    }

    private fun generateCircleVertices() {
        this.vertices = FloatArray((this.pointsCount + 1) * 3)
        var i = 3
        while (i < (this.pointsCount + 1) * 3) {
            val rad = i * 360 / this.pointsCount * 3 * (3.14 / 180)
            this.vertices[i] = Math.cos(rad).toFloat()
            this.vertices[i + 1] = Math.sin(rad).toFloat()
            this.vertices[i + 2] = 0f
            i += 3
        }
        this.componentsPerVertices = 3
    }

    override fun assignByValue(other: GLGeometry) {
        super.assignByValue(other)
        val otherCircle = other as GLCircle
        this.pointsCount = otherCircle.pointsCount
        this.generateCircleVertices()
        this.generateVertex()
    }
}
